apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,542 @@
+                "sg-148b4a5e",
+            ],
+            "assignPublicIp": "ENABLED"
+        }}
+    }},
+    "schedulingStrategy":"REPLICA",
+    "deploymentController":{{
+        "type": "ECS"
+    }},
+}}
+"""
+
+update_service_args = """
+{{
+    "cluster":"gsoc2019",
+    "service":"{service_name}",
+    "desiredCount":num_of_tasks,
+    "taskDefinition":"{task_def_arn}",
+    "forceNewDeployment":{forceNewDeployment}
+}}
+"""
+
+delete_service_args = """
+{{
+    "cluster": "gsoc2019",
+    "service": "{service_name}",
+    "force": False
+}}
+"""
+
+
+def client_token_generator():
+    """
+    Returns a 32 characters long client token to ensure idempotency with create_service boto3 requests.
+
+    Parameters: None
+
+    Returns:
+    str: string of size 32 composed of digits and letters
+    """
+
+    client_token = ''.join(random.choices(string.ascii_letters + string.digits, k=32))
+    return client_token
+
+
+def register_task_def_by_challenge_pk(client, queue_name, challenge):
+    """
+    Registers the task definition of the worker for a challenge, before creating a service.
+
+    Parameters:
+    client (boto3.client): the client used for making requests to ECS.
+    queue_name (str): queue_name is the queue field of the Challenge model used in many parameters fof the task def.
+    challenge (<class 'challenges.models.Challenge'>): The challenge object for whom the task definition is being registered.
+
+    Returns:
+    dict: A dict of the task definition and it's ARN if succesful, and an error dictionary if not
+    """
+    container_name = "worker_{}".format(queue_name)
+    image = "{AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-{env}-worker:latest".format(AWS_ACCOUNT_ID=aws_keys["AWS_ACCOUNT_ID"], env=ENV)
+    AWS_DEFAULT_REGION = aws_keys["AWS_REGION"]
+    log_group = "{}_logs".format(queue_name)
+
+    task_role_arn = TASK_ROLE_ARN
+    execution_role_arn = TASK_EXECUTION_ROLE_ARN
+
+    if task_role_arn and execution_role_arn:
+        definition = task_definition.format(queue_name=queue_name, task_role_arn=task_role_arn,
+                                            execution_role_arn=execution_role_arn, container_name=container_name,
+                                            image=image, AWS_DEFAULT_REGION=AWS_DEFAULT_REGION, env=ENV,
+                                            challenge_pk=challenge.pk, log_group=log_group, **DJANGO_SETTINGS_DICT)
+        definition = eval(definition)
+        if not challenge.task_def_arn:
+            try:
+                response = client.register_task_definition(**definition)
+                if (response["ResponseMetadata"]["HTTPStatusCode"] == HTTPStatus.OK):
+                    task_def_arn = response["taskDefinition"]["taskDefinitionArn"]
+                    challenge.task_def_arn = task_def_arn
+                    challenge.save()
+                return response
+            except ClientError as e:
+                logger.info(e)
```

Comment:
It should be `logger.exception()` here, no? 

Reference: https://docs.python.org/3/library/logging.html#logging.Logger.exception